Makefile.genspec

```diff
@@ -14,23 +143,80 @@ usage:
 	@echo "platforms"
 	@echo "Use target PLATFORM to make for a specific platform"
 
-ONE_GENERATED_FILE = cmake-mutatees.txt
+ONE_GENERATED_FILE = tuples tuples.gz cmake-mutatees.txt .gitignore
 ALL_GENERATED_FILES = $(foreach dir,$(SUBDIRS),$(ONE_GENERATED_FILE:%=$(dir)/%))
 
 gen-all: $(ALL_GENERATED_FILES)
 
 gen-clean:
 	-rm -f $(ALL_GENERATED_FILES)
 
+## gzip
+## -f force compression
+## -c keep tuples around for comparison
+## -n don't embed timestamp, so identical contents have identical compressed
+## This makes a rebuild not be a git change, unless actual changes occur.
+
+GPROLOG=gprolog
+GZIP=gzip
+MKDIR=mkdir
+
 $(SUBDIRS:%=%/tuples): %/tuples: $(PROLOG_FILES)
-	cd $(TO_SPEC); gprolog --entry-goal "['spec-grouped.pl']" \
+	cd $(TO_SPEC); \
+	$(MKDIR) -p `dirname ../../$@` ; \
+	$(GPROLOG)	\
+		--entry-goal "['$(SPEC)']" \
 		--entry-goal "test_init('$*')" \
 		--entry-goal "write_tuples('../../$@', '$*')" \
-		--entry-goal "halt" ; gzip -f ../../$@
+		--entry-goal "halt" ; \
+	$(GZIP) -f -n -c ../../$@ >../../$@.gz
 
 
 $(SUBDIRS:%=%/cmake-mutatees.txt): %/cmake-mutatees.txt: $(PYTHON_FILES) %/tuples
 	python -c "import sys; import os; os.environ['PLATFORM'] = '$*'; sys.path.append('$(TO_SPEC)'); import generate ; generate.generate('$*')"
 
 $(SUBDIRS:%=%): %:%/cmake-mutatees.txt
 
+## Generate common .gitignore files across platforms.
+## Basically Unix, except for the NT platforms
+
+## XXX could use filter/filter-out, or put this in a script, leave it for now.
+
+$(SUBDIRS:%=%/.gitignore): Makefile.genspec
```

Comment:
Splitting these by platform is overkill. I would just make a "master" .gitignore and copy it into the new dirs.